peartree/toolkit.py

```diff
@@ -69,8 +70,7 @@ def get_nearest_node(df_orig: pd.DataFrame,
                                  lng2=xs)
 
     # Calculate the final results to be returned
-    nearest_node = str(distances.idxmin())
-    nn_dist = distances.loc[nearest_node]
-
-    # Returna as tuple
-    return (nearest_node, nn_dist)
+    # Filter out nodes outside connection threshold and self (distance = 0)
+    nearest_nodes = distances[(distances > 0.0) & (distances < connection_threshold)]
```

Comment:
Need to "think this through." Why does the distance have to be greater than 0? There could be a stop that has the same identical point on two different GTFS feeds.

Reply:
Yeah you have a point. I vaguely remember thinking about doing it properly through id and then just cutting a corner since i would have to change the function parameters to pass the point and id instead of just the point to do a proper comparison... and then i got lazy :)

I can fix it later this week. Let me know.

Reply:
Thanks for the response. I am issuing another PR right now that will account for this!

Reply:
Great! To fully answer your previous question it wouldn't be sufficient to compare the point based on coordinates - those could match for two different feeds. It has to be compared based on the full name including the name prefix you've added i think.